Content.Shared/Containers/ItemSlot/ItemSlotsFillComponent.cs

```diff
@@ -0,0 +1,17 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Containers.ItemSlots;
+
+/// <summary>
+/// Fills an entity's itemslots at MapInit.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(ItemSlotsFillSystem))]
+public sealed partial class ItemSlotsFillComponent : Component
+{
```

Comment:
I don't think this is needed, you can just use the existing `ContainerFillComponent`

Reply:
this is better:
1. it validates slots whereas container doesnt care about slots
2. containerfill wont stop you writing multiple items in the same slot this you physically cant do that

Reply:
> it validates slots whereas container doesnt care about slots

ContainerFill also logs errors when a container doesn't exist.

> containerfill wont stop you writing multiple items in the same slot this you physically cant do that

There are tests that will start failing if you attempt to spawn multiple entities into one slot, so its not really an issue.

IMO its better to avoid unnecessarily duplicating this functionality.

Reply:
> whether or not that container belongs to a container slot doesn't matter.
it does matter
say i only specify the container named Thing and have ContainerFill for it, ContainerFill will work even though there isnt ItemSlots. this is a very easy to catch mistake, even if it might be extremely rare

and of course it is less typing than ContainerFill :trollface:

Reply:
> say i only specify the container named Thing and have ContainerFill for it, ContainerFill will work even though there isnt ItemSlots

That situation shouldn't error though? If the entity does have a container, then theres nothing wrong with container fill filling it? There's no need to add a new component type just so that you can assert that an entity has some other component

Reply:
an actual useful example would be machines, they have a container for parts. you wouldnt want to accidentally fill that instead of the desired itemslot, so this makes it completely impossible

Reply:
@ElectroJr so is this a hard requirement i really like the convenience of it :trollface:

Reply:
I'm also not a fan of having 3 different fill components that do the same thing.